src/charm.py

```diff
@@ -266,6 +284,26 @@ def __init__(self, *args):
         # Action handlers
         observe(self.on.show_proxied_endpoints_action, self._on_show_proxied_endpoints)  # type: ignore
 
+    @property
+    def _forward_auth_requirer_config(self) -> RequirerConfig:
```

Comment:
Since the provider has a dataclass, I'd prefer the requirer to have it as well for it to be coherent. I also don't think the requirer config will be subject to change once this PR is finalized.

Would `ForwardAuthRequirerConfig` be better?

Reply:
`ForwardAuthRequirerConfig` is clearer, but before we take that path: it's fine for the lib to use dataclasses. The question is should they leak outside of the lib. It's very pythonic to have kwargs.